test/testers.jl

```diff
@@ -263,4 +265,29 @@ end
         frule_test(iterfun, (x, ẋ))
         rrule_test(iterfun, randn(), (x, x̄))
     end
+
+    @testset "test quaternion non-standard scalar" begin
+        function FiniteDifferences.to_vec(q::Quaternion)
+            function Quaternion_from_vec(q_vec)
+                return Quaternion(q_vec[1], q_vec[2], q_vec[3], q_vec[4])
+            end
+            return [q.s, q.v1, q.v2, q.v3], Quaternion_from_vec
+        end
+
+        function ChainRulesCore.frule((_, Δq), ::typeof(quatfun), q)
+            ∂q = Quaternion(Δq)
+            return quatfun(q), Quaternion(∂q.v3, 2 * ∂q.v1, 3 * ∂q.s, 4 * ∂q.v2)
+        end
+
+        function ChainRulesCore.rrule(::typeof(quatfun), q)
+            function quatfun_pullback(ΔΩ)
+                ∂Ω = Quaternion(ΔΩ)
+                return (NO_FIELDS, Quaternion(3 * ∂Ω.v2, 2 * ∂Ω.v1, 4 * ∂Ω.v3, ∂Ω.s))
+            end
+            return quatfun(q), quatfun_pullback
+        end
+
+        q = quatrand()
```

Comment:
Should we define `rand_tangent(:: Quaternion)` in this package also?
@willtebbutt  do you have plans around further advancing `rand_tangent` ?

Reply:
I do not currently -- not sure that there's much to do beyond integrating it in with `ChainRulesTestUtils` in some way or another and continuing to add new methods where necessary.